### PR TITLE
[BUILD] add shellcheck

### PR DESCRIPTION
**Pull request by bmarwell**
* WHY? => Because of bashisms in this PR: https://github.com/apache/maven-wrapper/pull/65

* TODO: maybe change the output format? => https://github.com/koalaman/shellcheck/wiki/Integration#machine-parsable-output

* Blocked by: https://issues.apache.org/jira/browse/INFRA-24032 (enable shellcheck action)

---

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MWRAPPER) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MWRAPPER-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MWRAPPER-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [X] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [X] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [X] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [X] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [X] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Review by rmannibucau — APPROVED** [review]

**Review by slawekjaranowski — APPROVED** [review]

**Comment by bmarwell** [comment]
Draft b/c waiting for infra

**Comment by bmarwell** [comment]
@rmannibucau will now fail with:

```
./maven-wrapper-distribution/src/resources/mvnwDebug:35:68: note: Use $(...) notation instead of legacy backticks `...`. [SC2006]
./maven-wrapper-distribution/src/resources/only-mvnwDebug:24:3: note: Not following: ./mvnw was not specified as input (see shellcheck -x). [SC1091]
```

I think the first one is good, the second one needs some refining (ie exclude SC1091).

**Review by rmannibucau — APPROVED** [review]

**Comment by bmarwell** [comment]
Will add another PR which fixes the failed checks

File: .github/workflows/shellcheck-posix.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    env:
+      SHELLCHECK_VERSION: '0.9.0'
+      SHELLCHECK_OPTS: '-e SC2009'
+    steps:
+      - uses: actions/checkout@v3
+      - name: download
+        run: |
+          curl --silent --fail --show-error --retry 2 --retry-delay 1 --connect-timeout 5 --location --url "https://github.com/koalaman/shellcheck/releases/download/v${SHELLCHECK_VERSION}/shellcheck-v${SHELLCHECK_VERSION}.linux.x86_64.tar.xz" -o shellcheck-v${SHELLCHECK_VERSION}.tar.xz
+          tar xvf "shellcheck-v${SHELLCHECK_VERSION}.tar.xz"
+          rm -rf "shellcheck-v${SHELLCHECK_VERSION}.tar.xz"
```
Comment (pzygielo) [inline .github/workflows/shellcheck-posix.yml:37]: Is that `-r` necessary?

Comment (bmarwell) [reply .github/workflows/shellcheck-posix.yml:37]: no it is not. Will remove it.

**Pull request merged**